libvmi/core.c

```diff
@@ -565,13 +565,13 @@ vmi_init_custom(
     } else if (VMI_CONFIG_GHASHTABLE == config_mode) {
 
         char *name = NULL;
-        unsigned long domid = VMI_INVALID_DOMID;
+        uint32_t domid = VMI_INVALID_DOMID;
         GHashTable *configtbl = (GHashTable *)config;
         gpointer idptr = NULL;
 
         name = (char *)g_hash_table_lookup(configtbl, "name");
         if(g_hash_table_lookup_extended(configtbl, "domid", NULL, &idptr)) {
```

Comment:
This is kind of general question about the use of `VMI_CONFIG_GHASHTABLE`, but is there a definitive place other than looking at the internal code for someone to find out what the expected types of the values should be?